src/modules/ekf2/EKF2.cpp

```diff
@@ -354,12 +354,37 @@ void EKF2::Run()
 
 					// Validate the ekf origin status.
 					_ekf.getEkfGlobalOrigin(origin_time, latitude, longitude, altitude);
-					PX4_INFO("New NED origin (LLA): %3.10f, %3.10f, %4.3f\n", latitude, longitude, static_cast<double>(altitude));
+					PX4_INFO("New NED origin (LLA): %3.10f, %3.10f, %4.3f", latitude, longitude, static_cast<double>(altitude));
 				}
 			}
 		}
 	}
 
+	if (_vehicle_local_position_sub.updated()) {
+		vehicle_local_position_s lpos;
+
+		if (_vehicle_local_position_sub.update(&lpos)) {
+
+			if (lpos.ref_timestamp > 0
+			    && lpos.ref_timestamp > _ekf.global_origin().getProjectionReferenceTimestamp()
```

Comment:
Actually even direct timestamp comparison could be wrong in SITL where multiple instances might update in the same step (same timestamp).

Reply:
if ekf ref has been updated but not yet published in `local position` topic (for example update by vehicle command just above this bloc) we will have `lpos.ref_timestamp < _ekf.global_origin().getProjectionReferenceTimestamp()` and we don't want to set reference back to the previous one.
Would this be ok : ? 
```suggestion
			    && lpos.ref_timestamp >= _ekf.global_origin().getProjectionReferenceTimestamp()
```
I am afraid that in the future an ekf could change its ref (for example on a late init or anything else) so it will no be reset to the same ref as the others...

Another solution could be to not check the timestamp at all but not apply this on the selected instance.

Reply:
Here is a new proposal without timestamp check